Gradle plug-in no longer works w/o the `:mergeDiktatReports` task

As of version 1.2.2, the Gradle plug-in requires the `:mergeDiktatReports` task to be present, even for single-module projects.

This behaviour shouldn't apply in the following cases:

 * single-module projects;
 * projects where `reporter` is set to anything but `sarif` (e.g.: `plain`).

## Observed behavior

```console
C:\project> gradle --continue diktatCheck
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':diktatCheck'.
> Could not create task ':mergeDiktatReports'.
   > Could not create task of type 'SarifReportMergeTask'.
      > Could not generate a decorated class for type SarifReportMergeTask.
         > kotlinx/serialization/SerializationException

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
```

## Steps to Reproduce

Try to run `gradle diktatCheck` or `./gradlew diktatCheck` for either this [template project](https://github.com/0x6675636b796f75676974687562/diktat-demo-gradle), or any of the Gradle projects under `examples/`.

## Environment information
* diktat version: 1.2.2
* build tool (maven/gradle): Gradle
* link to a project: [0x6675636b796f75676974687562/diktat-demo-gradle](https://github.com/0x6675636b796f75676974687562/diktat-demo-gradle)

--- a/diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/tasks/SarifReportMergeTask.kt
+++ b/diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/tasks/SarifReportMergeTask.kt
@@ -17,6 +17,7 @@ import org.gradle.api.tasks.PathSensitive
 import org.gradle.api.tasks.PathSensitivity
 import org.gradle.api.tasks.TaskAction
 import org.gradle.api.tasks.TaskExecutionException
+import org.gradle.api.tasks.VerificationTask
 import kotlinx.serialization.SerializationException
 import kotlinx.serialization.decodeFromString
 import kotlinx.serialization.encodeToString
@@ -25,7 +26,7 @@ import kotlinx.serialization.json.Json
 /**
  * A task to merge SARIF reports produced by diktat check / diktat fix tasks.
  */
-abstract class SarifReportMergeTask : DefaultTask() {
+abstract class SarifReportMergeTask : DefaultTask(), VerificationTask {
     /**
      * Source reports that should be merged
      */
@@ -89,13 +90,13 @@ internal fun Project.configureMergeReportsTask(diktatExtension: DiktatExtension)
             reportMergeTask.output.set(mergedReportFile)
         }
     }
-    val reportMergeTaskTaskProvider = rootProject.tasks.named(MERGE_SARIF_REPORTS_TASK_NAME, SarifReportMergeTask::class.java) { reportMergeTask ->
+    rootProject.tasks.withType(SarifReportMergeTask::class.java).configureEach { reportMergeTask ->
         if (isSarifReporterActive(createReporterFlag(diktatExtension))) {
             getOutputFile(diktatExtension)?.let { reportMergeTask.input.from(it) }
             reportMergeTask.shouldRunAfter(tasks.withType(DiktatJavaExecTaskBase::class.java))
         }
     }
     tasks.withType(DiktatJavaExecTaskBase::class.java).configureEach { diktatJavaExecTaskBase ->
-        diktatJavaExecTaskBase.finalizedBy(reportMergeTaskTaskProvider)
+        diktatJavaExecTaskBase.finalizedBy(rootProject.tasks.withType(SarifReportMergeTask::class.java))
     }
 }

